dev/Materials/Acrylic/AcrylicBrush.cpp

```diff
@@ -419,39 +419,36 @@ winrt::Color AcrylicBrush::GetEffectiveLuminosityColor()
 // The tintColor passed into this method should be the original, unmodified color created using user values for TintColor + TintOpacity
 winrt::Color GetLuminosityColor(winrt::Color tintColor, winrt::IReference<double> luminosityOpacity)
 {
-    // To create the Luminosity blend input color, we're taking the TintColor input, converting to HSV, and clamping the V between these values
+    Rgb rgbTintColor = RgbFromColor(tintColor);
+
+    // If luminosity opacity is specified, just use the values as is
+    if (luminosityOpacity != nullptr)
+    {
+        return ColorFromRgba(rgbTintColor, std::clamp(luminosityOpacity.GetDouble(), 0.0, 1.0));
+    }
+
+    // To create the Luminosity blend input color without luminosity opacity,
+    // we're taking the TintColor input, converting to HSV, and clamping the V between these values
```

Comment:
I'd prefer you put the remainder of this function in an else block, I think it makes it easier to read at a glance than having to recognize the early return

Reply:
I am personally more a fan of the "return early" pattern, but sure, I will change it.

Reply:
You are still returning early but this puts the logic that is now in the else block at the correct indentation level. before this change its easier to look at the code that is now in the else block and assume it will be run regardless of whether or not the above if statement was executed or not.  Now it is obvious from the indentation level that it will only be run if we do not meet the if statement.

---
In reply to: [386686086](https://github.com/microsoft/microsoft-ui-xaml/pull/2017#discussion_r386686086) [](ancestors = 386686086)

Reply:
Oh haha, I see. So your preferred way would be have a variable and we either set it to the "normal" luminosity brush or run the calculations and return our saved variable at the end of the function? If you want I can still change that to only have one return at the end.